### PR TITLE
internal/v4: implement API endpoint for bundle diagram.svg

### PR DESCRIPTION
**Pull request by rogpeppe**
This requires some QA behind a real web server to
verify that the generated SVGs actually work!

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/454/

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/456/

File: internal/v4/api_test.go [hunk 1]
```diff
@@ -773,6 +775,154 @@ func (s *APISuite) TestServeExpandId(c *gc.C) {
...
+			break
+		}
+		c.Assert(err, gc.IsNil)
+		for what, f := range need {
+			if f(tok) {
+				delete(need, what)
+			}
+		}
+	}
+	c.Assert(need, gc.HasLen, 0, gc.Commentf("body:\n%s", body))
```
Comment (mhilton) [inline internal/v4/api_test.go:857]: Nice

**Comment by mhilton** [comment]
:+1: Looks good

**Comment by mhilton** [comment]
I've QA'd this and it generates pictures fine. We might want to consider how we deal with missing data as having no icon is a bit confusing. See http://localhost:8080/v4/bundle/transcode-orange-box-0/diagram.svg for an example.

**Comment by frankban** [comment]
Yeah we have a plan for missing icons, our current card is about caching icon contents so that we can decide whether to show default ones when generating the diagram.

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**